test/gw-ci/CMakeLists.txt

```diff
@@ -1,11 +1,220 @@
+      else()
+        list(APPEND TEST_DEPENDS "${test_prefix}_gdas_analcalc_${FULL_CYCLE}")
+        list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_eupd_${FULL_CYCLE}")
+      endif()
+    elseif("${task_name}" STREQUAL "enkfgdas_esnowrecen")
+      list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_epmn_${HALF_CYCLE}")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_snowanl_${FULL_CYCLE}")
+    elseif("${task_name}" STREQUAL "enkfgdas_esfc")
+      if("${pslot}" STREQUAL "C96C48_ufs_hybatmDA")
+        list(APPEND TEST_DEPENDS "${test_prefix}_gdas_analcalc_${FULL_CYCLE}")
+        list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_atmensanlfinal_${FULL_CYCLE}")
+      elseif("${pslot}" STREQUAL "C96C48_hybatmaerosnowDA")
+        list(APPEND TEST_DEPENDS "${test_prefix}_gdas_analcalc_${FULL_CYCLE}")
+        list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_eupd_${FULL_CYCLE}")
+        list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_esnowrecen_${FULL_CYCLE}")
+      else()
+        list(APPEND TEST_DEPENDS "${test_prefix}_gdas_analcalc_${FULL_CYCLE}")
+        list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_eupd_${FULL_CYCLE}")
+      endif()
+    elseif("${task_name}" STREQUAL "enkfgdas_fcst")
+      list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_ecmn_${FULL_CYCLE}")
+      list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_esfc_${FULL_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_prepoceanobs")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_fcst_${HALF_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_marinebmat")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_fcst_${HALF_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_marineanlinit")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_fcst_${HALF_CYCLE}")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_prepoceanobs_${FULL_CYCLE}")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_marinebmat_${FULL_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_marineanlvar")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_marineanlinit_${FULL_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_marineanlchkpt")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_marineanlvar_${FULL_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_marineanlfinal")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_marineanlchkpt_${FULL_CYCLE}")
+    elseif("${task_name}" STREQUAL "gdas_marineanlletkf")
+      list(APPEND TEST_DEPENDS "${test_prefix}_enkfgdas_fcst_${HALF_CYCLE}")
+      list(APPEND TEST_DEPENDS "${test_prefix}_gdas_prepoceanobs_${FULL_CYCLE}")
+    else()
+      list(APPEND TEST_DEPENDS "${test_prefix}")      
+    endif()
+  endif()    
+
+  # Set cycle
+  if(${is_full_cycle})
+    set(cycle ${FULL_CYCLE})
+  else()
+    set(cycle ${HALF_CYCLE})
+  endif()
+  
+  # Add Task
+  set(test_name ${test_prefix}_${task_name}_${cycle})
+  message(STATUS "preparing ${subtask_names} for ${test_prefix} ctest")
+  ecbuild_add_test(TARGET ${test_name}
```

Comment:
ok so you basically have this here, so I suggest reproducing it in the aforementioned CMakeLists.txt